src/main/java/org/nrnb/gsoc/enrichment/tasks/FilterEnrichmentTableTask.java

```diff
@@ -1,113 +1,121 @@
+            gravity = 1.0
+    )
+    public ListMultipleSelection<String> evidenceCodes;
+
+    private CyApplicationManager applicationManager;
+    private EnrichmentCytoPanel enrichmentPanel;
+    private CyNetwork network;
+    private CyTable filteredEnrichmentTable;
+    private CyServiceRegistrar registrar;
+    private List<TermSource> categoryFilter = TermSource.getValues();
+
+    //public boolean removeOverlapping = false;
+
+    //public BoundedDouble overlapCutoff = new BoundedDouble(0.0, 0.5, 1.0, false, false);
+
+    public FilterEnrichmentTableTask(final CyServiceRegistrar registrar, EnrichmentCytoPanel panel) {
+        this.registrar = registrar;
+        applicationManager = registrar.getService(CyApplicationManager.class);
+        this.network = applicationManager.getCurrentNetwork();
+        this.enrichmentPanel = panel;
+        final CyColumn evidenceColumn = ModelUtils.getEnrichmentTable(registrar, network, TermSource.ALL.getTable())
+                .getColumn(EnrichmentTerm.colTermID);
+        System.out.println(evidenceColumn.getValues(String.class).size());
+        final Set<String> evidenceCodeList = new TreeSet<>(evidenceColumn.getValues(String.class));
+        this.evidenceCodes = new ListMultipleSelection<>(new ArrayList<>(evidenceCodeList));
+    }
 
-	//public BoundedDouble overlapCutoff = new BoundedDouble(0.0, 0.5, 1.0, false, false);
 
-	public FilterEnrichmentTableTask(final CyServiceRegistrar registrar, EnrichmentCytoPanel panel) {
-    this.registrar = registrar;
-		applicationManager = registrar.getService(CyApplicationManager.class);
-		this.network = applicationManager.getCurrentNetwork();
-		this.enrichmentPanel = panel;
-	}
+    @Override
+    public void run(TaskMonitor monitor) throws Exception {
+        monitor.setTitle("Filter Enrichment table");
 
+        List<TermSource> categoryList = categories.getSelectedValues();
+        HashSet<String> evidenceList = new HashSet<>(evidenceCodes.getSelectedValues());
+        if (enrichmentPanel == null) {
+            CySwingApplication swingApplication = registrar.getService(CySwingApplication.class);
+            CytoPanel cytoPanel = swingApplication.getCytoPanel(CytoPanelName.SOUTH);
+            if (cytoPanel.indexOfComponent("org.nrnb.gsoc.enrichment") != -1)
+                enrichmentPanel = (EnrichmentCytoPanel) cytoPanel.getComponentAt(
+                        cytoPanel.indexOfComponent("org.nrnb.gsoc.enrichment"));
+            else return;
+        }
+        EnrichmentTableModel tableModel = enrichmentPanel.getTableModel();
+        if (tableModel == null) {
+            monitor.showMessage(TaskMonitor.Level.ERROR, "Unable to find enrichment table!");
+        }
+        if (!categoryList.isEmpty()) tableModel.filter(categoryList);
+        if (!evidenceList.isEmpty()) tableModel.filterByEvidenceCode(evidenceList);
+    }
 
+    @Override
+    @SuppressWarnings("unchecked")
+    public <R> R getResults(Class<? extends R> clzz) {
```

Comment:
Suggestion: We could use a more descriptive variable name instead `clzz`

Reply:
It is a section of code already present. Intellij formated the file itself so this is shown as a new section.

Reply:
Would do all formatting in a separate PR for the project.